numpy/lib/tests/test_shape_base.py

```diff
@@ -104,6 +106,21 @@ def test_broadcast(self):
         put_along_axis(a, ai, 20, axis=1)
         assert_equal(take_along_axis(a, ai, axis=1), 20)
 
+    def test_invalid(self):
+        """ Test invalid inputs """
+        a_base = np.array([[10, 30, 20], [60, 40, 50]])
+        indices = np.array([[0], [1]])
+        values = np.array([[2], [1]])
+
+        # sanity check
+        a = a_base.copy()
+        put_along_axis(a, indices, values, axis=0)
+
+        # invalid indices
+        a = a_base.copy()
+        assert_raises(ValueError, put_along_axis, a, indices, values, axis=None)
```

Comment:
This test will also pass before the PR. 
```suggestion
        with assert_raises(ValueError) as exc:
            put_along_axis(a, indices, values, axis=None)
        assert "single dimension" in str(exc.exception)
```